### DISCUSSION TITLE
ESPN_2 and SWID Credentials

### DISCUSSION THREAD

**Discussion by slmroz** [cwendt94/espn-api#150] [ESPN API] [answered]
Hi there.
First of all - great to see ESPN API project as ESPN abandoned the idea years ago.
As I can see now - they introduced additional recapcha authentication so now it's impossible to get access to private leagues using the userid and pass.
What's the easiest way to find the SWID and ESPN_S2 values? Is it stored in cookies alter the login? Does it stay the same with next sessions?

**Comment by dtcarls** [comment, marked as answer]
8/20/25 update: Updating this answer with an easier method. The below still fully works. There is now a [chrome extension](https://chromewebstore.google.com/detail/espn-private-league-setup/bjmalaafoepfooflcnhjejnopgefjgia?authuser=0&hl=en) that will pull this from your cookies that is a bit easier than above. The extension is [open source](https://github.com/dtcarls/ESPNExtension) if you want to take a look at the code first.

You can find these two values after logging into your espn fantasy football account on espn's website. (Chrome Browser) Right click anywhere on the website and click inspect option. From there click Application on the top bar. On the left under Storage section click Cookies then http://fantasy.espn.com. From there you should be able to find your swid and espn_s2 variables and values.
It remains the same through different sessions.

**Reply by cwendt94** [reply]
Thanks @dtcarls! That reminds me I need to update the Wiki with this information.

**Reply by slmroz** [reply]
Thank you very much for your answer.

**Reply by rysull23** [reply]
@dtcarls Do you think any code could be written to gather this info on someone else's behalf as part of a 'log on' process?

**Reply by rysull23** [reply]
...I imagine you could somehow trigger this extension and capture it that way. Where to begin!?

**Reply by dtcarls** [reply]
It is probably possible, but it would likely require using selenium or some browser then grab the cookies associated with it. All that being said, it would be much more heavy weight than a python library for making API calls which would fall outside the scope of this library.

**Reply by rysull23** [reply]
agreed. I wonder if you can somehow package the code from the open source program and pass it in to auto-execute as part of a python program. Otherwise, the only way someone can benefit from this API in their league is if they can provide you their own cookie information. It would be important to figure this out so you can tap into other's leagues once they've granted permission.
